protocol/contracts/templegold/DaiGoldAuction.sol

```diff
@@ -0,0 +1,339 @@
+     * @return Bool if allowing deposit
+     */
+    function canDeposit() external view override returns (bool) {
+        return epochs[_currentEpochId].isActive();
+    }
+
+    /**
+     * @notice Get auction configuration
+     * @return Auction configuration
+     */
+    function getAuctionConfig() external override view returns (AuctionConfig memory) {
+        return auctionConfig;
+    }
+
+    /**
+     * @notice Get claimable amount for current epoch
+     * @dev For current epoch, function will return claimable at current time. This can change with more user deposits
+     * @param depositor Address to check amount for
+     * @return Claimable amount
+     */
+    function getClaimableAtCurrentEpoch(address depositor) external override view returns (uint256) {
+        return getClaimableAtEpoch(depositor, _currentEpochId);
+    }
+
+    /**
+     * @notice Get claimable amount for an epoch
+     * @dev For current epoch, function will return claimable at current time. This can change with more user deposits
+     * @param depositor Address to check amount for
+     * @param epochId Epoch id
+     * @return Claimable amount
+     */
+    function getClaimableAtEpoch(address depositor, uint256 epochId) public override view returns (uint256) {
+        if (claimed[depositor][epochId]) { return 0; }
+        uint256 bidTokenAmount = depositors[depositor][epochId];
+        if (bidTokenAmount == 0 || epochId > _currentEpochId) { return 0; }
+        EpochInfo memory info = epochs[epochId];
+        return bidTokenAmount.mulDivRound(info.totalAuctionTokenAmount, info.totalBidTokenAmount, false);
+    }
+    
+    /**
+     * @notice Recover auction tokens for last but not started auction. 
+     * Any other token which is not Temple Gold can be recovered too at any time
+     * @dev For recovering Temple Gold, Epoch data is deleted and leftover amount is added to nextAuctionGoldAmount.
+     * so admin should recover total auction amount for epoch if that's the requirement
+     * @param token Token to recover
+     * @param to Recipient
+     * @param amount Amount to auction tokens
+     */
+    function recoverToken(
+        address token,
+        address to,
+        uint256 amount
+    ) external override onlyElevatedAccess {
+        if (to == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        if (amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+
+        if (token != address(templeGold)) {
+            emit CommonEventsAndErrors.TokenRecovered(to, token, amount);
+            IERC20(token).safeTransfer(to, amount);
+            return;
+        }
+
+        // auction started but cooldown pending
+        uint256 epochId = _currentEpochId;
+        EpochInfo storage info = epochs[epochId];
+        if (info.startTime == 0) { revert InvalidOperation(); }
+        if (info.isActive()) { revert AuctionActive(); }
+        if (info.hasEnded()) { revert AuctionEnded(); }
```

Comment:
nit:
```suggestion
        if (block.timestamp >= info.startTime) revert InvalidOperation();
```
I beliee some of these requirements are redundant. 

```solidity
        if (info.startTime == 0) { revert InvalidOperation(); }  // require: startTime > 0
        if (info.isActive()) { revert AuctionActive(); }  // require: timestamp < start  ||  timestamp >= end
        if (info.hasEnded()) { revert AuctionEnded(); }  // require: timestamp < end
```

The last two requirements can be condensed, so that the three requirements are in reality:
```
// require(startTime > 0  &&  timestamp < startTime);
```
However, since `startTime` can never be negative,  the requirement `timestamp < startTime`  has already implicit that `stratTime > 0` holds . So actually, the second condition is enough, and the three requirements can be replaced by:

```solidity
require(block.timestamp < info.startTime);
```
or:
```solidity
if (block.timestamp >= info.startTime) revert InvalidOperation();
```

Reply:
unless you want to keep the three different errors for user experience reasons ...